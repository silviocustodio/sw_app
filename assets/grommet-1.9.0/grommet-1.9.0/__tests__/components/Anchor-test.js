// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import Anchor from '../../src/js/components/Anchor';

import FakeIcon from '../mocks/FakeIcon';

describe('Anchor', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Anchor href='test' />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct primary=true rendering', () => {
    const component = renderer.create(
      <Anchor href='test' primary={true} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct disabled=true rendering', () => {
    const component = renderer.create(
      <Anchor href='test' disabled={true} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct icon rendering', () => {
    const component = renderer.create(
      <Anchor icon={<FakeIcon />} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct icon as a child rendering', () => {
    const component = renderer.create(
      <Anchor><FakeIcon /></Anchor>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct text as a child rendering', () => {
    const component = renderer.create(
      <Anchor>test</Anchor>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct icon and label rendering', () => {
    const component = renderer.create(
      <Anchor icon={<FakeIcon />} label='test' />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('calls onClick after rendering', () => {
    const onAnchorClick = jest.fn();
    const component = renderer.create(
      <Anchor href='test' onClick={onAnchorClick} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.props.onClick();
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(onAnchorClick).toBeCalled();
  });
  it('preserves id after rendering', () => {
    const component = renderer.create(
      <Anchor id='test' href='test' />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
