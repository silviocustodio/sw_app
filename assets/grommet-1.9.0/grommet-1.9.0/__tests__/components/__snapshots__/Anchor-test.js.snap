// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor calls onClick after rendering 1`] = `
<a
  aria-label={undefined}
  className="grommetux-anchor"
  href="test"
  onClick={[Function]}
/>
`;

exports[`Anchor calls onClick after rendering 2`] = `
<a
  aria-label={undefined}
  className="grommetux-anchor"
  href="test"
  onClick={[Function]}
/>
`;

exports[`Anchor has correct default options 1`] = `
<a
  aria-label={undefined}
  className="grommetux-anchor"
  href="test"
  onClick={[Function]}
/>
`;

exports[`Anchor has correct disabled=true rendering 1`] = `
<a
  aria-label={undefined}
  className="grommetux-anchor grommetux-anchor--disabled"
  href="test"
  onClick={[Function]}
/>
`;

exports[`Anchor has correct icon and label rendering 1`] = `
<a
  aria-label={undefined}
  className="grommetux-anchor grommetux-anchor--animate-icon grommetux-anchor--icon grommetux-anchor--icon-label"
  href={undefined}
  onClick={[Function]}
>
  <svg
    height="24px"
    role="img"
    version="1.1"
    viewBox="0 0 24 24.3616"
    width="24px"
  />
  test
</a>
`;

exports[`Anchor has correct icon as a child rendering 1`] = `
<a
  aria-label={undefined}
  className="grommetux-anchor grommetux-anchor--animate-icon grommetux-anchor--icon"
  href={undefined}
  onClick={[Function]}
>
  <span
    className="grommetux-anchor__icon"
  >
    <svg
      height="24px"
      role="img"
      version="1.1"
      viewBox="0 0 24 24.3616"
      width="24px"
    />
  </span>
</a>
`;

exports[`Anchor has correct icon rendering 1`] = `
<a
  aria-label={undefined}
  className="grommetux-anchor grommetux-anchor--animate-icon grommetux-anchor--icon"
  href={undefined}
  onClick={[Function]}
>
  <span
    className="grommetux-anchor__icon"
  >
    <svg
      height="24px"
      role="img"
      version="1.1"
      viewBox="0 0 24 24.3616"
      width="24px"
    />
  </span>
</a>
`;

exports[`Anchor has correct primary=true rendering 1`] = `
<a
  aria-label={undefined}
  className="grommetux-anchor grommetux-anchor--animate-icon grommetux-anchor--icon grommetux-anchor--primary"
  href="test"
  onClick={[Function]}
>
  <svg
    aria-label="link next"
    className="grommetux-control-icon grommetux-control-icon-link-next grommetux-control-icon--responsive"
    height="24px"
    role="img"
    version="1.1"
    viewBox="0 0 24 24"
    width="24px"
  >
    <path
      d="M2,12 L22,12 M13,3 L22,12 L13,21"
      fill="none"
      stroke="#000"
      strokeWidth="2"
    />
  </svg>
</a>
`;

exports[`Anchor has correct text as a child rendering 1`] = `
<a
  aria-label={undefined}
  className="grommetux-anchor"
  href={undefined}
  onClick={[Function]}
>
  test
</a>
`;

exports[`Anchor preserves id after rendering 1`] = `
<a
  aria-label={undefined}
  className="grommetux-anchor"
  href="test"
  id="test"
  onClick={[Function]}
/>
`;
